clib/gc_chooser.h

```diff
@@ -22,5 +22,6 @@ void initialize_gc_option(void); /* Select the wanted GC using envvar `NIT_GC_OP
 
 void gc_register_finalizer(void*); /* Tag a pointer for finalization */
 void gc_finalize(void*, void*); /* Finalize a pointer, implemented in the generated code. */
+void object_destroy_callback(void*, void*); /* call into an object finalizer to record some traces. */
```

Comment:
I've put it there because traces.h is include in a specific structure in lttng's files. this declaration is not expected in those lttng files. Nothing else than macros can be written in the traces.h provider.